.gitignore

```diff
@@ -121,3 +121,6 @@ package-lock.json
 
 # lighthouse build directory
 .lighthouseci/
+
```

Comment:
Probably worth calling out this tool in the root CONTRIBUTING doc and linking to its homepage.
https://github.com/ni/nimble/blob/main/CONTRIBUTING.md#repository-overview

And if there is anything devs need to know about using it you could add a section to that doc with more instructions. But I can't think of anything specific to call out, so might not be necessary.

Reply:
Added a line to the CONTRIBUTING

Reply:
This is a bit confusing to me. You said you reverted the Wireit files from GitHub Desktop, but didn't that include all of the .wireit directories? You say:
> To get back to a good state, I needed to manually find + delete all the .wireit directories from the various subfolders within the Nimble repo

I guess if you delete just *some* of the files from the .wireit directories, then you'll be in a bad state. Is that what happened?

Reply:
> Going forward this scenario will probably be less likely (once our branches in-flight also use wireit), but still a command/docs on purging the cache could be useful.

It might be useful to have that generally as wireit lets the cache grow indefinitely: https://github.com/google/wireit#local-caching

They suggest the command `rm -rf .wireit/*/cache` which is not cross-platform.

Instead we can opt-in each package that uses rimraf to a "clear-cache" npm script that can be called from the root with `npm run clear-cache --workspaces --if-present`. We can use the `rimraf` npm package to make the script cross-platform.

So we can document how to clean your cache if it's getting large and maybe broadcast to the team about using that command for while branches are temporarily out of date from main (I doubt folks will look in the docs for that note).

Reply:
Adding the clear-cache commands like Milan outlined seems useful, and would be one way to get out of the state I was in. (I guess the brute-force `git clean -fxdq` is another).

I was able to reproduce my scenario.
- Start from this branch
- (Clean everything with `git clean -fxdq`, and `npm install` again)
- `npm run build`
- Switch to the `main` branch
- Undo all remaining changes that show up in the GitHub Desktop pending changes window. This was the file list I had: [wireit - files to undo.txt](https://github.com/ni/nimble/files/11619370/wireit.-.files.to.undo.txt)
- After doing that I printed out a list of the remaining local .wireit files I had via PowerShell (`Get-ChildItem .\*\.wireit -Recurse | ForEach-Object {  Write-Host $_.fullname }`):
[wireit - remaining local files.txt](https://github.com/ni/nimble/files/11619374/wireit.-.remaining.local.files.txt)

So the issue is that there were still .wireit cache changes partially remaining locally, as you said. Looking at the file paths it looks like it's because we already gitignore'd `dist` folders, so `**/.wireit/**/dist/*` were still there locally (but didn't show up as pending changes). So the cache was in a bad state / partially still there, since all the cached files with `dist` in their path remained.

Reply:
The proposals above seem good to me. I also hit this issue when switching branches.

If adding the clear-cache command is much work (I doubt it, but just in case) I'd also be ok with adding docs that mention the unbounded cache growth and suggest running `git clean -fdx` periodically. I don't think the switching branches issue will be a problem for long.

Reply:
Yea documenting running `git clean -fdx` periodically make sense to me. Just do a nice deep scrub once in a while.

Reply:
I added a "Local repo maintenance" section to the CONTRIBUTING where I document these commands. I added `clear-cache` as well as `clear-wireit`, which wipes the entire `.wireit` directory -- not just the cache.